bridge/client/app/_components/ktb-evaluation-details/ktb-evaluation-details.component.ts

```diff
@@ -785,13 +785,20 @@ export class KtbEvaluationDetailsComponent implements OnInit, OnDestroy {
 
   highlightHeatmap(): void {
     if (this._selectedEvaluationData && !this.isInvalidated) {
+      this.comparedIndicatorResults = [];
+      const secondaryHighlightIndexes = this._selectedEvaluationData?.data.evaluation?.comparedEvents?.map(
+        (eventId) => {
+          const eventIndex = this._heatmapSeries[0]?.data.findIndex((e) => e.evaluation?.id === eventId);
+          this.comparedIndicatorResults.push(
+            this._heatmapSeries[0]?.data[eventIndex].evaluation?.data.evaluation?.indicatorResults ?? []
+          );
+          return eventIndex;
```

Comment:
Currently, we are loading all evaluations, right? So the index can't technically be `-1` or `undefined`. Nevertheless, I would consider this case and exclude not found evaluations. This would also be future proof then if we ever decide to not load all evaluations.

Reply:
Yea, true. Actually we had that check, but since we are loading all evaluations, I thought that would be good without the check, but still I think it might happen that some evaluations are not loaded (e.g. we do have a heatmap pageSize). Restructured this a little bit and moved all related code to the `setSecondaryHighlight` method.